benchmark/benchmark.go

```diff
@@ -30,3 +54,101 @@ func GetSubject() (string, error) {
 	}
 	return subject, nil
 }
+
+// Wrapper reports the benchmarking metrics of targetFunc to testing.B.
+func Wrapper(b *testing.B, targetFunc func(), cleanupFunc func()) {
+	metricsSum := Metrics{}
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		b.StartTimer()
+		metrics, err := measureMetrics(targetFunc)
+		if err != nil {
+			b.Fatal("Error measuring metrics:", err)
+			return
+		}
+		b.StopTimer()
+		metricsSum = addMetrics(metricsSum, metrics)
```

Comment:
If you change the function as `func (m1 *Metrics) addMetrics(m2 Metrics) {`
you can simply write as `metricsSum.Add(metrics)`